#181414758 feature(login):create user login functionality

Procech20 opened the pull request:
### What does this PR do?
Create login functionality on the app.

### Description of Task to be completed?
- Create user login route
- Tested the route
- Documented the route

### How should this be manually tested?
To test the changes manually:
* clone repo on your local machine
* switch to develop branch  by typing the command ```git checkout ft-login-181414758```
* Run: ```yarn install```: to install all dependencies
* Create .env file
* Create a ```.env``` and the variables as mentioned in the ```.env.example``` file.
* Open the host address with the endpoint ``/api/v2/auth/login`` using an HTTP client such as ``postman``` or any other.
* Make a ``POST`` request using an ``email`` and ``password``.

### What to expect after running the commands:
* _On successful login, A user will get a message that says ``User login successful`` with a status of ``200``, and will get a token to be used to make other requests to other endpoints that requires sign in_

* _If the user doesn't exist, you'll a message that says: ``Invalid Email or password`` with the status of: ``404``._

### Any background context you want to provide?
- N/A

``` Useful links ```:
- N/A

### What are the relevant pivotal tracker stories?
* https://www.pivotaltracker.com/story/show/181414758

### Screenshots (if appropriate)
- N/A

### Questions:
- N/A

Robertishimwe commented:
Every time I am trying login with email and password i am getting this message.
<img width="628" alt="prof" src="https://user-images.githubusercontent.com/81079680/158454644-b105132a-394d-4ec0-b627-b6fff10f6c70.PNG">

it seems like there is no authentication taking place

ericus123 committed:
feat(Login):create user login functionality to the app

- create login functionality with email & password
- documentation of the endpoint with swagger
- test endpoint with mocha & chai

[Delivers #181414758]
